mood-analysis.rb

```diff
@@ -19,7 +19,57 @@ def analyze_mood(words)
   return ":-|"
 end
 
+def strip_punctuation(string)
+  string.gsub!(/[^0-9A-Za-z]/, ' ')
+end
+
+def happy_days(list_of_moods)
+  # how many tweets until there have been 3 happy days?
+  number_of_happy_days = 0
+  moods_read = 0
+  list_of_moods.each do |mood|
+    # THANK YOU RUBY KOANS!!!
+    break moods_read if number_of_happy_days == 3
+    number_of_happy_days += 1 if mood == ":-)"
+    moods_read += 1
+  end
+  if number_of_happy_days >= 3
+    return "It takes #{moods_read} entries for 3 happy days to occur"
+  else
+    return "Only #{number_of_happy_days} happy day(s)...bummer!"
+  end
+end
```

Comment:
This method is fine, but I would have liked to see it return the smallest piece of information possible. In this case, a fixnum representing the number of days it takes to hit 3 happy days. Then we would have the ability to use this method in different ways instead of being tied to this specific text / use case.